api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	//
```

Comment:
we need to add some explicit comments here, saying - "Rules are executed in order, if a rule matches, the `action` is executed and further rules are skipped"

Reply:
We may want to handle rule priority DENY > ALLOW, like [Istio Authz does](https://istio.io/latest/docs/reference/config/security/authorization-policy/).

Reply:
do you think how Istio defines it is easy to understand ?

I find it very hard, which is why I am a +1 for "first match in rules list wins"

Reply:
Think again and agree, `First match wins` is more intuitive.

Reply:
@arkodg This may be able to be achieved with [matcher API](https://www.envoyproxy.io/docs/envoy/latest/xds/type/matcher/v3/matcher.proto#envoy-v3-api-msg-xds-type-matcher-v3-matcher). However, Envoy doc says it's WIP and not recommended to be used in production.

Reply:
id still suggest us using it, we have used the other alpha APIs in the project already

Reply:
Yeah, Macher API is much more flexible and solves all the concerns raised in the discussion. Let's go with it.